conformance/packages/conformance-tests/src/resolver/dns/scenarios.rs

```diff
@@ -43,7 +43,7 @@ fn can_resolve() -> Result<()> {
 #[ignore]
 #[test]
 fn nxdomain() -> Result<()> {
-    let needle_fqdn = FQDN("unicorn.nameservers.com.")?;
+    let needle_fqdn = FQDN::TEST_DOMAIN.push_label("unicorn");
```

Comment:
Personally I would rename the method to `prepend_label` to be clearer. Imho `push_*` looks close to `Vec::push`, which appends.

Reply:
I sort of agree but also the hierarchy of domain names flows from right to left so "push happens on the left side" does not seem strange to me considering that.

perhaps I'll rename this to `.subdomain(label)` since it creates a subdomain below the callee. also `push_label` sounds a bit like a mutator (`&mut self`) but this method is not one.

Reply:
> perhaps I'll rename this

(not in this PR though)